File: frontend/src/types/processing.ts
export type OutputType = "image" | "chart" | "value" | "table";

export interface ProcessingOutput {
  name: string;
  type: OutputType;
  data: unknown;
  caption?: string | null;
  filename?: string | null;
}

export interface ProcessingResult {
  success: boolean;
  processingTime: number;
  outputs: ProcessingOutput[];
  metadata: Record<string, unknown>;
  warnings: string[];
  error?: string | null;
}

export interface ApiErrorPayload {
  error: {
    code: string;
    message: string;
  };
}

export interface StudentInfo {
  name: string;
  rollNumber: string;
  usn: string;
  semester: string;
  section: string;
  date: string;
}

File: backend/app/schemas/processing.py
"""Pydantic request/response models for the processing endpoint."""
from __future__ import annotations

from typing import Any, Literal

from pydantic import BaseModel


class OutputSchema(BaseModel):
    name: str
    type: Literal["image", "chart", "value", "table"]
    data: Any
    caption: str | None = None
    filename: str | None = None


class ProcessingResponse(BaseModel):
    success: bool
    processingTime: float
    outputs: list[OutputSchema]
    metadata: dict[str, Any] = {}
    warnings: list[str] = []
    error: str | None = None


class ErrorDetail(BaseModel):
    code: str
    message: str


class ErrorResponse(BaseModel):
    error: ErrorDetail
